src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let result = sourceString.split(';');
+
+  result = result.filter((item) => item !== '');
```

Comment:
The filter operation here is not correctly removing empty strings. It should check for strings that are not just empty but also those that may contain only whitespace. You can use a more robust check like `.filter(item => item.trim() !== '')`.